graspologic/nominate/VNviaSGM.py

```diff
@@ -0,0 +1,205 @@
+import numpy as np
+from ..match import GraphMatch as GMP
+from sklearn.base import BaseEstimator
+
+
+class VNviaSGM(BaseEstimator):
+    """
+    This class implements vertex nomination via seeded graph matching
+
+    Parameters
+    ----------
+    h: int
+        distance between voi used to create induced subgraph on `G_1`
+
+    ell: int
+        distance from seeds to other verticies to create induced subgraphs on `G_1`
+
+    R: int
+        Number of restarts for soft seeded graph matching algorithm
+
+    a_inds: 1d-array
+        Array of indices in the induced subgraph of A that will be matched to `b_inds`
+
+    b_inds: 1d-array
+        Array of indices in the induced subgraph of B that will be matched to `a_inds`
+
+    n_seeds_used: int
+        Number of seeds passed in `seedsA` that occured in the induced subgraph about `voi`
+
+    nomination_list: list
+        List of 2 tuples of the format nomination_list = [(j, p_val)] where p_val is the
+        probability that the voi matches to node j in graph B (sorted by descending
+        probability)
+
+    Attributes
+    ----------
+    P : 2d-array, square
+        The probability array that describes the probability `P[i, j]` that row `i` in `a_inds`
+        matches `j` in `b_inds`
+
+    corr : 1d-array
+        The matrix that maps values from `a_inds` to `b_inds`
+
+
+    References
+    ----------
+    .. [1] Patsolic, HG, Park, Y, Lyzinski, V, Priebe, CE. Vertex nomination via seeded graph matching. Stat Anal Data
+    Min: The ASA Data Sci Journal. 2020; 13: 229– 244. https://doi.org/10.1002/sam.11454
+    """
+
+    def __init__(
+        self,
+        h=1,
+        ell=1,
+        R=100
+    ):
+        self.h = h
+        self.ell = ell
+        self.R = R
+
+    def fit(self, voi, A, B, seedsA, seedsB):
+        """
+        Parameters
+        ----------
+
+        voi: int
+            Vertices of interest (voi)
+
+        A: 2d-array, square
+            Adjacency matrix of `G_1`, the graph where voi is known
+
+        B: 2d-array, square
+            Adjacency matrix of `G_2`, the graph where voi is not known
+
+        seedsA: list
+            Seeds associated to adjacency matrix A
+
+        seedsB: list
+            Seeds associated to adjacency matrix B `len(seedsA)==len(seedsB)`
```

Comment:
for  `fit` and `fit_predict`, the parameters don't really conform to sklearn norms, though I'm not really sure what the best way to do that here would be, or how much they're ok with deviation from sklearn conventions. It might be fine as is.